egov-searcher/src/main/java/org/egov/SearchApplicationRunnerImpl.java

```diff
@@ -45,27 +51,84 @@
     @Override
     public void run(final ApplicationArguments arg0) throws Exception {
     	try {
-				logger.info("Reading yaml files......");			
-			    readFiles();			
+				logger.info("Reading yaml files......");
+			    readFiles();
 			}catch(Exception e){
 				logger.error("Exception while loading yaml files: ",e);
 			}
     }
-    
+
+	//file types to be resolved have to be passed as comma separated types.
+	public List<String> resolveAllConfigFolders(List<String> listOfFiles, Boolean resolveFolderCheck, String fileTypesToResolve){
+	List<String> ymlUrlList = new ArrayList<String>();
+	String[] fileTypes = fileTypesToResolve.split("[,]");
+
+	if(resolveFolderCheck==false){
+		return listOfFiles;
+	}else{
+		for(int i=0;i<listOfFiles.size();i++){
+			String[] fileName = listOfFiles.get(i).split("[.]");
+				if(searchArray(fileName[fileName.length - 1], fileTypes)){
+					ymlUrlList.add(listOfFiles.get(i));
+
+				}else{
+					ymlUrlList.addAll(FolderIterator(listOfFiles.get(i),fileTypes));
+				}
+
+		}
+		return ymlUrlList;
+	}
+	}
+
+	public Boolean searchArray(String exten,String[] fileTypes){
+    	if(exten==null)	return false;
+    	for(String extension: fileTypes){
+    		if(extension.equals(exten)){
+    			return true;
+			}
+		}
+    	return false;
+	}
+
+	public List<String> FolderIterator(String baseFolderPath, String[] fileTypes) {
+		File folder = new File(baseFolderPath);
+		File[] listOfFiles = folder.listFiles();
+		List<String> configFolderList = new ArrayList<String>();
+
+		for (int i = 0; i < listOfFiles.length; i++) {
+			log.info("File " + listOfFiles[i].getName());
+			File file = listOfFiles[i];
+			String name = file.getName();
+			String[] fileName = name.split("[.]");
+			if (searchArray(fileName[fileName.length - 1], fileTypes)) {
```

Comment:
You don't need a additional function. You can use something like `fileTypes.contains(fileName[fileName.length - 1])`